### DISCUSSION TITLE
How to use View Transition API page transitions for specific scenarios, and enable Nuxt Page Transitions for everything else

### DISCUSSION THREAD

**Discussion by toddpadwick** [nuxt/nuxt#26199] [Questions]
I have a website, which makes use of [Nuxt Vue Page Transitions](https://nuxt.com/docs/getting-started/transitions#page-transitions) for the majority of its page changes. However, there are specific cases, such as clicking on the Case Studies further down [my websites homepage](https://www.functionandform.co.uk/), in which I have opted to make use of the new [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API) to animate the listed case study state to become the case study hero seamlessly (only works in Chromium based browsers as of 03/24). To achieve this, I have added a query param – `?native=true` – on the case studies link, and with the following middleware, I listen to the existence of such a query param to turn off Nuxt Vue Page Transitions for that route change.
```
export default defineNuxtRouteMiddleware(to => {
  if (import.meta.server || !document.startViewTransition) { return } // if browser doesn't support native view transitions API, use nuxt page transitions for everything
  if (to.query?.native) { 
    // if view transitions are supported and the native query param is present in the destination route path, disable Nuxt Vue Transitions and allow native view transition API to do its thing
    to.meta.pageTransition = false
    to.meta.layoutTransition = false
  }
})
```
This successfully turns off Nuxt Vue Page transitions and allows the native Chromium _View Transitions API_ to take over. However, if you click on a link in the nav such as about or work etc from that case study you have just visited, you'll notice the Nuxt Vue Page Transitions remain disabled. So, this indicates an issue where by somehow, my middleware code prevents the page transitions to be reenabled for the a subsequent page. So with this problem in mind, I have two questions:

1. How do I re-enable Nuxt Vue Page transitions for the following page change after I have disabled them in middleware?
2. Is there a better, cleaner way of achieving this, without query params? seems a tad clunky?

**Comment by felixfavour** [comment]
@toddpadwick - nothing much to add to this thread, just that I love your application of the view transition API on your website. It's beautiful.

The load time might annoy a few, but it's worth it for me, lol.

**Reply by toddpadwick** [reply]
Hi @felixfavour  i didnt see this notification until just now. Thank you for your kind words!

**Comment by rowild** [comment]
@toddpadwick Really awesome site! Bookmarked!

**Comment by michaelpumo** [comment]
Surely you would re-enable them by just setting them explicitly like how you disable them?

```typescript
export default defineNuxtRouteMiddleware(to => {
  if (import.meta.server || !document.startViewTransition) {
    // if browser doesn't support native view transitions API, use nuxt page transitions for everything
	to.meta.pageTransition = true
	to.meta.layoutTransition = true
  }
  else if (to.query?.native) { 
    // if view transitions are supported and the native query param is present in the destination route path, disable Nuxt Vue Transitions and allow native view transition API to do its thing
    to.meta.pageTransition = false
    to.meta.layoutTransition = false
  }

  return
})
```
